verify/verifier.go

```diff
@@ -132,9 +132,21 @@ func (v *Verifier) VerifyCommitment(expectedCommit *common.G1Commitment, blob []
 
 	expectedX := &fp.Element{}
 	expectedX.Unmarshal(expectedCommit.X)
+
+	// map coordinates to G1 and ensure they are on the curve
+	xAffine := bn254.MapToG1(*expectedX)
+	if !xAffine.IsOnCurve() {
+		return fmt.Errorf("expected x is not on the curve: %x", expectedX.Marshal())
```

Comment:
"expected x" feels like a very testy variable name. Maybe change to `claimedCommitment` everywhere instead?

Reply:
but commitment is a constitution of (x,y) field elements.... trying to think of better name here